web/security/csrf.py

```diff
@@ -0,0 +1,116 @@
+# encoding: utf-8
+
+import hashlib, hmac
+from os import urandom
+from binascii import hexlify, unhexlify
+
+from web.security.predicate import Predicate
+
+try:
+    from hmac import compare_digest
+except ImportError:
+    def compare_digest(a, b):
+        return a == b
+
+
+log = __import__('logging').getLogger(__name__)
+
+
+class CSRFError(ValueError):
+	pass
+
+
+class CSRFPasses(Predicate):
+    def __init__(self, *predicates):
+        pass
+    
+    def __call__(self, context):
+        if context.request.method in ('POST', 'PUT', 'DELETE'):
+            # Probably need to make the var name configurable somehow
+            token = context.request.POST['csrftoken']
```

Comment:
Should be configurable, and also this causes a KeyError if the csrf token isn't sent in the request in the first place.